src/Platform/Microsoft.Testing.Extensions.VSTestBridge/ObjectModel/ContextAdapterBase.cs

```diff
@@ -61,4 +73,117 @@ protected ContextAdapterBase(ICommandLineOptions commandLineOptions)
 
         return adapterSpecificTestCaseFilter;
     }
+
+    private void HandleFilter(ITestExecutionFilter? filter, string? filterFromRunsettings, string? filterFromCommandLineOption)
+    {
+        // No filters at all, we can return immediately as there is nothing to do.
+        if (filter is null or NopFilter && filterFromRunsettings is null && filterFromCommandLineOption is null)
```

Comment:
Nit: for readability (feel free to ignore)

```suggestion
        if (filter is null or NopFilter 
            && filterFromRunsettings is null 
            && filterFromCommandLineOption is null)
```